### PR TITLE
Make PER overview partially editable

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/views/CountryPreparedness/index.tsx [hunk 1]
```diff
@@ -249,146 +261,192 @@ export function Component() {
...
+                        data={assessmentStats.ratingByArea}
+                    />
+                </Container>
+            )}
+            {hasPer && (
+                <Container
+                    heading={strings.previousPerAssessmentHeading}
+                    withHeaderBorder
+                >
+                    Chart not available
```
Comment (samshara) [inline src/views/CountryPreparedness/index.tsx]: Shouldn't this come from translations

File: src/views/CountryPreparedness/index.tsx [hunk 2]
```diff
@@ -249,146 +261,192 @@ export function Component() {
...
+                    heading={strings.priorityComponentToBeStrengthenedHeading}
+                    childrenContainerClassName={styles.priorityComponentsContent}
+                    withHeaderBorder
+                >
+                    <BarChart
+                        data={prioritizationStats.componentsWithRating}
+                        keySelector={numericIdSelector}
+                        labelSelector={(component) => component.details.title}
+                        valueSelector={(component) => component.rating?.value ?? 0}
+                    />
```
Comment (samshara) [inline src/views/CountryPreparedness/index.tsx]: Let's use defined functions

File: src/components/ChartAxes/index.tsx [hunk 3]
```diff
@@ -29,6 +29,7 @@ interface Props<X, Y> {
         width: number;
         height: number;
     };
+    showBoundRects?: boolean,
```
Comment (tnagorra) [inline src/components/ChartAxes/index.tsx]:
```suggestion
    boundRectsShown?: boolean,
```

File: src/components/ChartAxisY/index.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+function ChartAxisY(props: Props) {
+    const {
+        ticks,
+        chartMargin,
+        chartBounds,
+        chartInnerOffset,
+    } = props;
+
+    const yAxisTickHeight = Math.max(
```
Comment (tnagorra) [inline src/components/ChartAxisY/index.tsx:34]: Let's add NOTE why dividing by zero is not problematic here.

File: src/views/CountryPreparedness/index.tsx [hunk 5]
```diff
@@ -73,6 +80,7 @@ export function Component() {
     });
 
     const perId = latestPerResponse?.results?.[0]?.id;
+    const prevAssessmentRatings = latestPerResponse?.results?.[0]?.assessment_ratings;
```
Comment (tnagorra) [inline src/views/CountryPreparedness/index.tsx:84]: Why do we take first element? Let's add a NOTE

File: src/views/CountryPreparedness/index.tsx [hunk 6]
```diff
@@ -205,7 +221,7 @@ export function Component() {
                 ),
                 (answerList) => ({
                     id: answerList[0].answer,
-                    label: formAnswerMap[answerList[0].answer] ?? '-',
+                    label: `${formAnswerMap[answerList[0].answer]} ${answerList.length}`,
```
Comment (tnagorra) [inline src/views/CountryPreparedness/index.tsx:225]: Are we sure that the mapping will not return an undefined value

File: src/views/CountryPreparedness/index.tsx [hunk 7]
```diff
@@ -240,155 +257,245 @@ export function Component() {
...
+                        label={strings.focalPointEmailLitle}
+                        value={overviewResponse?.ns_focal_point_email}
+                        strongValue
+                    />
+                    <div className={styles.contactContainer}>
+                        <Link
+                            to={perTeamEmail}
+                            variant="secondary"
+                        >
+                            Contact PER Team
```
Comment (tnagorra) [inline src/views/CountryPreparedness/index.tsx:357]: Let's translate

File: src/views/CountryPreparedness/index.tsx [hunk 8]
```diff
@@ -240,155 +257,245 @@ export function Component() {
...
+                                key={component.details.id}
+                            >
+                                <Heading level={5}>
+                                    {`${component.details.component_num}: ${component.details.title}`}
+                                </Heading>
+                                <ProgressBar
+                                    value={component.rating?.value}
+                                    totalValue={5}
+                                    title={isDefined(component.rating) ? (
+                                        `${component.rating?.value} - ${component.rating?.title}`
```
Comment (tnagorra) [inline src/views/CountryPreparedness/index.tsx]: Let's make sure we don't pass undefined in string literals

File: src/views/CountryPreparedness/index.tsx [hunk 9]
```diff
@@ -240,155 +257,245 @@ export function Component() {
...
+                                    ) : '0 - Not Reviewed'}
```
Comment (tnagorra) [inline src/views/CountryPreparedness/index.tsx]: Let's translate

File: src/views/CountryPreparedness/PreviousAssessmentChart/index.tsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+                {chartData.points.map(
+                    (point) => (
+                        <circle
+                            className={styles.circle}
+                            key={point.key}
+                            cx={point.x}
+                            cy={point.y}
+                        >
+                            <title>
+                                {`Assessment: ${point.xValue}, Rating: ${point.yValue ?? '-'}`}
```
Comment (tnagorra) [inline src/views/CountryPreparedness/PreviousAssessmentChart/index.tsx:74]: Let's translate

File: src/views/CountryPreparedness/PreviousAssessmentChart/index.tsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+                />
+                <ChartAxisY
+                    chartBounds={chartBounds}
+                    chartMargin={chartData.chartMargin}
+                    chartInnerOffset={chartData.innerOffset}
+                    ticks={chartData.yAxisTicks}
+                />
+                {chartData.points && (
+                    <path
+                        d={getDiscretePathDataList(chartData.points)?.[0]}
```
Comment (tnagorra) [inline src/views/CountryPreparedness/PreviousAssessmentChart/index.tsx]: why only use first elem?

File: src/utils/per.ts [hunk 12]
```diff
@@ -1,11 +1,11 @@
...
-export const STEP_OVERVIEW = 1;
-export const STEP_ASSESSMENT = 2;
-export const STEP_PRIORITIZATION = 3;
-export const STEP_WORKPLAN = 4;
-export const STEP_ACTION = 5;
+export const PER_PHASE_OVERVIEW = 1;
+export const PER_PHASE_ASSESSMENT = 2;
+export const PER_PHASE_PRIORITIZATION = 3;
+export const PER_PHASE_WORKPLAN = 4;
+export const PER_PHASE_ACTION = 5;
```
Comment (tnagorra) [inline src/utils/per.ts]: Let;s use `satisfies`

File: src/utils/chart.ts [hunk 13]
```diff
@@ -68,7 +69,7 @@ export function getScaleFunction(
 }
 
 // TODO: Add test
-export function getBounds(numList: number[]) {
+export function getBounds(numList: UnsafeNumberList, zeroMin = false) {
```
Comment (tnagorra) [inline src/utils/chart.ts]: Let's remove zeroMin from this function

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Pull request merged**